examples/tidal_barrage/lagoon2d.py

```diff
@@ -91,7 +92,7 @@
 
 
 def update_forcings(t_new,):
-    tidal_elev.assign(Constant(tanh((t_new)/(4*3600.)) * amplitude * sin(omega * t_new)))
+    tidal_elev.assign(math.tanh((t_new)/(4*3600.)) * sin(omega * t_new) * amplitude)
```

Comment:
Uhhh, does this do compilation for every new `t_new` (or does `assign` no longer do code generation @connorjward)

Reply:
No, that's what I thought as well at first, but these are all just floats, so the entire thing should evaluate to a float. Actually, looking at this again tidal_elev should just be a `Constant`

Reply:
In the old world f.assign(float) would
compile code hardcoding the value of the float, but not sure what happens now

Reply:
Since it's a `Constant` we just assign directly to the `Global` ([source](https://github.com/firedrakeproject/firedrake/blob/master/firedrake/constant.py#L142)). And if this were a `Function`, since we don't do codegen any more, it makes no difference.